### ISSUE TITLE
Error thrown when increasing fee: "sendrawtransaction RPC error: {"code":-26,"message":"non-BIP68-final"}

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#4736]
This error is thrown sometimes when a user tries to increase the fee on a transaction. 

![image](https://github.com/leather-wallet/extension/assets/33360391/bea725cc-4f42-4b1e-bcae-449c5d377fe1)

It was seen the first time with this transaction when I tried to increase the fee on it a couple minutes after I broadcasted it:
https://mempool.space/testnet/tx/dae341077edf50483da1a88aabf02bf8f0572f39e0fa4e95e2a64757acce6ba8

For this transaction, the first two increases were no problem but on the third attempt the user got the error:
https://mempool.space/testnet/tx/376e79da9cae66f83d1a20ad417b83ab4399f3a18fc2eaa87c352c3a7f316864

The user from the previous transaction saw no issues with this one but they followed the same process as before.
https://mempool.space/testnet/tx/7a556809f0174a3b8963e7b93f2d83a8deb90a6095fff3ddf40b0e224469c6c3

BIP0068 standard
https://en.bitcoin.it/wiki/BIP_0068

Mentions of final:
![image](https://github.com/leather-wallet/extension/assets/33360391/f2ae52b0-1422-459c-af30-4453486c03e5)

This error was also reported four weeks ago for a mainnet transaction 
https://mempool.space/tx/b83be82f9e61c82f553e2fc2e2086b977477af3cb5e7942d76814f7c71a28bb0

I assumed the problem was related to the fact that the inputs were already spent and currently Leather doesn't support adding additional inputs to cover increased fees (or adjusting the primary output = the amount being sent).

**Comment by thinkchocolate** [comment]
Hey is there any solution to this error as I have today experienced this same error message when trying to increase fees on a transaction "code":-26, "message":"non-BIP68 final" - I have reached out to Leather wallet support, waiting a response, and so far cannot move the transaction along its been 9 painful hours...

**Comment by 314159265359879** [comment]
A user reporting a similar issue reported getting it when they tried to increase the fee on this transaction:
​https://mempool.space/tx/c01ad206490fe4b6e451dcf780ace3bd58d4f724fce65d84595b1428286be91e

![image](https://github.com/leather-wallet/extension/assets/33360391/b4e2224b-0a62-43f2-8043-c8c988fafee3)

Despite the wallet looking like it should be able to about double the current fee with the available ~20k sats (from 100 sats/vB to about 200 sats/vB) this error is thrown instead:

![image](https://github.com/leather-wallet/extension/assets/33360391/c76363eb-1982-464b-80fe-e6377a278db4)

**Comment by thinkchocolate** [comment]
My issue is now resolved the mem pool must have cleared the fees. Not clicking on any links too cautious, screen shots please.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by 314159265359879** [comment]
The user with this transaction also saw this non-bip68-final error come up when they tried to increase the fee for this transaction after it had been pending for almost a day:
https://mempool.space/tx/2e8ac71ea9b2561fa7a99a458b859debd5139fb70bb19338bd771bd397cd9f46

By the time I got to their support issue, it had already been confirmed for 18 hours. I am not sure if they did anything to progress it or it got included without additional effort. There is an expensive counterparty/stamp transaction in the CPFP chain which could explain why a much higher fee was needed to get it in (those have a high adjusted vsize).

For users it would be better if the error displayed "increasing the fee for this transaction is not supported yet"?

**Comment by markmhendrickson** [comment]
@314159265359879 should we convert this issue into an enhancement for better edge-case handling?

**Comment by 314159265359879** [comment]
@markmhendrickson would that result in a higher priority? :-)

Users like to have the ability to increase the fee on any transaction regardless of the structure. I do think it makes sense to see this as an enhancement to handle edge cases. Should we consider the same for [https://github.com/leather-wallet/extension/issues/4784](https://github.com/leather-wallet/extension/issues/4784#issuecomment-1933679629)?

**Comment by markmhendrickson** [comment]
Yep let's consider both issues enhancements. Can you upgrade their titles as well so they indicate the sort of cases in which increase fees need to work but don't yet? I anticipate we'll tackle these during approval UX redesign work.
